### DISCUSSION TITLE
rights to distribute files copyrighted by Dinkumware

### DISCUSSION THREAD

**Discussion by luisapace** [uNetworking/uWebSockets.js#338] [Q&A] [answered]
Hi, my team would like to use the package uWebSockets.js-18.3.0, and to approved that I've to verify all licenses.

I've analyzed this package and I've found the sentence "Copyright (c) by P.J. Plauger, licensed by Dinkumware, Ltd. ALL RIGHTS RESERVED" in the following binaries embedded in the code:

<img width="240" alt="image" src="https://user-images.githubusercontent.com/27818611/86140516-3a868300-baf1-11ea-95af-3926069cd415.png">

It does not appear that this open source package can just distribute files copyrighted by Dinkumware unless you have a license first. May I kindly ask you if you have a license to redistribute code from Dinkumware?

Thanks.

**Comment by ghost** [comment]
> Dinkumware is an American software company specializing in core libraries for C/C++, owned and operated by P. J. Plauger. It is based in Concord, Massachusetts.

> The company has provided the C++ Standard Library implementation that ships with Microsoft Visual C++ since 1996

What kind of internal deal Microsoft and Dinkumware have is up to them. Those binaries are the output of Microsoft Visual C++.

All windows binaries have that tag, even node.exe.

**Comment by ghost** [comment, marked as answer]
I found this:

https://devblogs.microsoft.com/cppblog/open-sourcing-msvcs-stl/

> Acknowledgements
The MSVC team would like to thank Dinkumware and P.J. Plauger for making this possible.

Where the LLVM Exception is extremely permissive:

https://spdx.org/licenses/LLVM-exception.html

> As an exception, if, as a result of your compiling your source code, portions of this Software are embedded into an Object form of such source code, you may redistribute such embedded portions in such Object form without complying with the conditions of Sections 4(a), 4(b) and 4(d) of the License.

So yes, we undoubtedly have rights to distribute these binaries. And if you think about it, the entire software ecosystem would implode if we didn't. (Literally all software ever compiled for Windows would be owned by Microsoft).

Alright

**Comment by luisapace** [comment]
Thanks a lot!

**Comment by ghost** [comment]
No problem, this was interesting to research anyways
